docs/src/chemiscope-sphinx.rst

```diff
@@ -0,0 +1,74 @@
+.. _chemiscope-sphinx:
+
+Chemiscope in `sphinx`
+======================
+
+The chemiscope python module also provides a `sphinx` extension that can process
+a custom RST directive into an interacting viewer, that can be embedded in the 
+generated HTML documentation.
+
+Configuration
+^^^^^^^^^^^^^
+
+Assuming a project is already configured to use `sphinx` to build its documentation,
+in order to be able to use the chemiscope directive one has to first
+add the `chemiscope.sphinx` extension to the `docs/conf.py` file:
+
+.. code-block:: python
+
+   extensions = ["chemiscope.sphinx"]
+
+Chemiscope directive
+^^^^^^^^^^^^^^^^^^^^
+
+In order to include a chemiscope viewer into a RST documentation page one can use a 
+``chemiscope`` directive. When compiled with ``sphinx-builder``, the directive will
+be translated into an HTML embedding of and interactive widget, that loads a JSON 
+file and displays it. The directive specifies two options:  ``filepath`` - that 
+indicates the path relative to the RST file where the chemiscope JSON is to be found, 
+and  ``mode`` that specifies the type of visualization to be used
+(``default``, ``structure``, or ``map``).
+
+For instance, to show a viewer combining structure and property panels you can 
+include the directive
+ 
+.. code-block:: rst
+
+    .. chemiscope::
+        :filepath: datasets/showcase.json.gz
+        :mode: default 
```

Comment:
Regarding syntax: I would take the path as the main argument to the directive, i.e.

```
.. chemiscope:: datasets/showcase.json.gz
    :mode: default 
```

and make `mode` default to "default" if not given. So this would be used like

```
.. chemiscope:: datasets/showcase.json.gz


.. chemiscope:: datasets/showcase.json.gz
    :mode: structure
```

Reply:
ok I agree this would be really neat - IDK how harder it'd be to write the custom directive and document it but yes if it's simple let's do this